Classes/Emogrifier.php

```diff
@@ -1303,9 +1303,9 @@ private function matchIdAttributes(array $match)
      */
     private function matchClassAttributes(array $match)
     {
-        return ($match[1] !== '' ? $match[1] : '*') . '[contains(concat(" ",@class," "),concat(" ","' .
+        return ($match[1] !== '' ? $match[1] : '*') . '[contains(concat(" ",normalize-space(@class)," "),concat(" ","' .
```

Comment:
It can't go in a local variable as its part of an xpath- its actually returning an expression to be evaluated in another context. The xpath evaluator should however optimize it away anyway.